fix: skip over preinstalled ENS snap when custom RPC is used for mainnet

mirceanis opened the pull request:
## **Description**

We now have a preinstalled snap that does the [ENS forward resolution in the Metamask-extension](https://github.com/MetaMask/metamask-extension/pull/26242)

However, because ENS requires a mainnet connection and there is no way for a snap to request a particular provider configuration from the wallet, the snap is forced to use a direct connection to Infura in certain corner cases.

This is a temporary situation, as there is a plan to [enhance the provider API for snaps](https://github.com/MetaMask/MetaMask-planning/issues/2915).

Until we have that better API we should respect the user's choice to not use Infura, so the compromise solution that was proposed is to disable the snap in those corner cases when the user is using a custom RPC for mainnet.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27367?quickstart=1)

## **Related issues**

fixes https://github.com/MetaMask/MetaMask-planning/issues/3339

## **Manual testing steps**

### Direct test
1. Configure a custom RPC URL for mainnet (e.g. https://eth.llamarpc.com)
2. Select that custom RPC URL
3. Switch networks to some other network (e.g. Linea)
4. Initiate a send flow and try to resolve an ENS name (e.g. `vitalik.eth`)
5. Observe the error `No resolution for domain provided.`
### Other cases are not affected
6.Switch networks to mainnet (keeping custom RPC selected)
7. Observe `vitalik.eth` resolving to an address
8. Select infura RPC for mainnet and select mainnet as a network
9. Observe `vitalik.eth` resolving to an address
10. While infura RPC is selected for mainnet, switch to a different network
11. Observe `vitalik.eth` resolving to an address from mainnet (with a warning :warning:)

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

https://github.com/user-attachments/assets/3d3881cf-a127-431d-9f48-89702ca57d0e

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

mirceanis committed:
fix: skip over preinstalled ENS snap when custom RPC is used for mainnet

fixes https://github.com/MetaMask/MetaMask-planning/issues/3339

Signed-off-by: Mircea Nistor <mirceanis@gmail.com>

Matt561 commented on ui/ducks/domains.test.ts:
```diff
@@ -0,0 +1,238 @@
+    jest.clearAllMocks();
+  });
+
+  it('should not skip snap for mainnet', async () => {
```
Minor: Our [contributor docs](https://github.com/MetaMask/contributor-docs/blob/main/docs/unit-testing.md#use-it-to-specify-the-desired-behavior-for-the-code-under-test) say to not use `should` in our test descriptions.

However, a quick global search in the repo shows that we have over three thousand tests breaking this rule.

![image](https://github.com/user-attachments/assets/56ea67ed-b661-4608-bded-7932bab71fba)

It's possible that the guidelines were created after these tests were written. 

I'll leave it up to you to change or leave as is! I thought I'd mention it because I received the same comment on a recent PR 😄

Matt561 reviewed (dismissed):
I left a very minor comment related to MetaMask contributor guidelines concerning test descriptions. 

Otherwise, LGTM! 💯

mirceanis committed:
test: follow coding guideline for unit tests

https://github.com/MetaMask/contributor-docs/blob/main/docs/unit-testing.md#examples
Signed-off-by: Mircea Nistor <mirceanis@gmail.com>
